src/spartan_http_handler.erl

```diff
@@ -0,0 +1,165 @@
+-include_lib("dns/include/dns.hrl").
+-include_lib("erldns/include/erldns.hrl").
+
+init(_Transport, Req, Opts) ->
+    {upgrade, protocol, cowboy_rest, Req, Opts}.
+
+rest_init(Req, Opts) ->
+    {ok, Req, Opts}.
+
+allowed_methods(Req, State) ->
+    {[<<"GET">>], Req, State}.
+
+content_types_provided(Req, State) ->
+    {[
+        {{<<"application">>, <<"json">>, []}, process}
+    ], Req, State}.
+
+process(Req, State = [version]) ->
+    Applications = application:which_applications(),
+    {spartan, _V, Version} = lists:keyfind(spartan, 1, Applications),
+    Body =
+        jsx:encode([
+            {<<"Service">>, <<"Spartan">>},
+            {<<"Version">>, list_to_binary(Version)}
+        ]),
+    {Body, Req, State};
+
+process(Req, State = [config]) ->
+    reply_halt(501, Req, State); % TODO
+
+process(Req, State = [hosts]) ->
+    {Host, Req0} = cowboy_req:binding(host, Req),
+    Query = #dns_query{name = Host, type = ?DNS_TYPE_A},
+    DNSMessage = #dns_message{
+        rd = true, ad = true,
+        qc = 1, questions = [Query],
+        adc = 1, additional = [#dns_optrr{}]},
+    #dns_message{answers = Answers} =
+        erldns_handler:handle(DNSMessage, {http, Host}),
+    Data =
+        [ [{<<"host">>, Name}, {<<"ip">>, ntoa(Ip)}]
+        || #dns_rr{name = Name, type = ?DNS_TYPE_A,
+                   data = #dns_rrdata_a{ip = Ip}} <- Answers],
+    {jsx:encode(Data), Req0, State};
+
+process(Req, State = [services]) ->
+    {Service, Req0} = cowboy_req:binding(service, Req),
+    Query = #dns_query{name = Service, type = ?DNS_TYPE_SRV},
+    DNSMessage = #dns_message{
+        rd = true, ad = true,
+        qc = 1, questions = [Query],
+        adc = 1, additional = [#dns_optrr{}]},
+    #dns_message{answers = Answers} =
+        erldns_handler:handle(DNSMessage, {http, Service}),
+    Data =
+        lists:map(fun (Record) ->
+            #dns_rr{
+                name = Name, type = ?DNS_TYPE_SRV,
+                data = #dns_rrdata_srv{
+                    port = Port, target = Target}
+            } = Record,
+            Json = [
+                {<<"host">>, Target},
+                {<<"ip">>, <<>>},
+                {<<"port">>, Port},
+                {<<"service">>, Name}
+            ],
+            try resolve(Target) of Ip ->
+                lists:keystore(<<"ip">>, 1, Json, {<<"ip">>, ntoa(Ip)})
+            catch throw:_ ->
+                lists:keydelete(<<"ip">>, 1, Json)
+            end
+        end, Answers),
+    {jsx:encode(Data), Req0, State};
+
+process(Req, State = [enumerate]) ->
+    reply_halt(501, Req, State); % TODO
+
+process(Req, State = [records]) ->
+    {{chunked, records_fun()}, Req, State}.
+
+records_fun() ->
+    fun (SendFun) ->
+        ok = SendFun("["),
+        ZonesV = erldns_zone_cache:zone_names_and_versions(),
+        Zones = [Z || {Z, _} <- ZonesV],
+        records_fun(SendFun, [], Zones, 0),
+        ok = SendFun("]")
+    end.
+
+records_fun(_SendFun, [], [], N) ->
```

Comment:
No. It's good style to give a name to unused variable.

http://learnyousomeerlang.com/starting-out-for-real#invariable-variables

```
Technically, variables can start with an underscore ('_') too, but by convention
their use is restricted to values you do not care about, yet you felt it was
necessary to document what it contains.
```